### PR TITLE
Project RecipesApp

### PR DESCRIPTION
**Pull request by dennis-every**
This is a pull request to merge the dev branch into the main branch after completing the following project requirements:

### Project requirements
- You should follow the layout of the wireframes provided. 
- You should personalize the rest of the design including colors, typographies, spacings, etc.
- Login page and registration page:
- Should be built with Devise.
- Food list (only for a group that has less than 3 members):
- Should display a list of food added by the logged-in user as in the wireframe (display also quantity of a given food).
- Should lead to a form that allows users to add new food.
- Recipes list:
- Should display a list of recipes created by the logged-in user as in the wireframe.
- Should lead to recipe details.
- If the user is the owner of the recipe, should allow the user to delete it.
- Public recipe list:
- Should display a list of all public recipes ordered by newest as in the wireframe.
- Should lead to recipe details.
- Recipe details:
- Should display a toggle button that allows for a recipe to be made public or private.
- If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
- If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- General shopping list view (only for a group that has less than 3 members):
- Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
- Should count the total food items and total price of the missing food.
- Make sure there are no N+1 queries happening.
- Create a navigation menu that allows users to open all of the pages you created.
- Write unit and integration tests

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
```
Comment (afanimartin) [inline app/controllers/application_controller.rb:2]: - Please initialize `cancancan` authorization gem by adding this line `load_and_authorize_resource`. You could also just add that to the specific controllers that need authorization

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @ashmalzahra, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- You manage to create the required controllers for `Food`, `Recipe`, `Ingredient` and `ShoppingList` with the necessary actions
- Tests added for models
- Views created for the respective controllers with links to the appropriate actions
- Repo looks professional and contains contributions from both students 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Find me on slack
### Martin Afani

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      </div>
+    <% end %>
+
+    <div class="text-center">
+      <%= f.submit "Log in", class: "px-4 py-2 bg-indigo-600 text-white rounded-md hover:bg-indigo-700" %>
+    </div>
+  <% end %>
+
+  <%= render "devise/shared/links" %>
+</div>
```
Comment (VICTOR-CHUKWU) [inline app/views/devise/sessions/new.html.erb:28]: - [ ] There is an error in your sign_in and sign_up page kindly fix
Screenshot from 2023-05-27 02-06-43

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi Team, 
## STATUS: CHANGES REQUIRED :recycle: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @dennis-every , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Well done working on the changes requested by the previous reviewer.  🎉 🎉 🎉 ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**